### PR TITLE
editorial changes

### PR DESCRIPTION
**Pull request by afrind**

### CODE REVIEW

**Comment by erickinnear** [comment]
:)

**Comment by erickinnear** [comment]
Since these two will conflict (#2/#3 with #5), do you want to merge this first and then I'll re-apply my changes on top?

File: draft-xie-bidirectional-messaging-latest.md [hunk 1]
```diff
@@ -41,49 +41,51 @@ HTTP/2 is the de facto application protocol in Internet. The optimizations
...
+interactions onto one HTTP/2 stream reintroduces head of line blocking. Also,
+application metadata is encapsulated into DATA frames, rather than HEADERS
+frames, making header compression is impossible. Further, user data is framed
+multiple times at different protocol layers, which offsets the wire efficiency
+of HTTP/2 binary framing. Take WebSocket over HTTP/2 as an example, user data is
+framed at the application protocol, WebSocket, and HTTP/2 layers. This not only
+introduces the overhead on the wire, but also complicates data processing.
+Finally, intermediaries have no visibility to user interactions layered on a
+single HTTP/2 stream, and lose the capability to collect telemetry metrics
+(e.g., time to the first/last byte of request and response) for services.
```
Comment (erickinnear) [inline draft-xie-bidirectional-messaging-latest.md:77]: 👍

**Comment by afrind** [comment]
Derp I didn’t even see your PRs.  I’ll defer to Woo on which order to apply.  I can rebase merge if needed.

File: draft-xie-bidirectional-messaging-latest.md [hunk 2]
```diff
@@ -41,49 +41,51 @@ HTTP/2 is the de facto application protocol in Internet. The optimizations
...
+
+These techniques also pose new operational challenges to intermediaries. Because
+all traffic from a user's session is encapsulated into one HTTP/2 stream, this
+stream can last a very long time. Intermediaries may take long time to drain
```
Comment (erickinnear) [inline draft-xie-bidirectional-messaging-latest.md:81]:
```suggestion
stream can last a very long time. Intermediaries may take a long time to drain
```

File: draft-xie-bidirectional-messaging-latest.md [hunk 3]
```diff
@@ -111,17 +113,18 @@ facilitate XStreams' intermediary traversal.
 A new HTTP/2 stream called eXtended stream (XStream) is introduced for
 exchanging user data bidirectionally. An XStream is opened by an XHEADERS frame,
 and **MAY** be continued by CONTINUATION and DATA frames. XStreams can be
-initiated by either clients or servers. Another difference from a regular
-stream, an XStream **MUST** be associated with an open RStream. In this way,
-XStreams can be routed according to their RStreams by intermediaries and
-servers. XStream **MUST NOT** be associated with any other XStream, or any
-closed RStream. Otherwise, it cannot be routed properly.
+initiated by either clients or servers. Unlike regular a stream, an XStream
```
Comment (erickinnear) [inline draft-xie-bidirectional-messaging-latest.md]:
```suggestion
initiated by either clients or servers. Unlike a regular stream, an XStream
```

File: draft-xie-bidirectional-messaging-latest.md [hunk 4]
```diff
@@ -148,9 +151,10 @@ from client to server, either endpoint can initiate an XStreams to its peer.
 A client can multiplex RStreams, XStreams and regular HTTP/2 streams into one
 single HTTP/2 connection. Beyond that, all the XStreams associated with the same
 RStream form a logical stream group, and are routed to the same endpoint. This
-enables clients to access different services without initiating new connections.
-As shown in the following diagram, the client can exchange data with PubSub, RPC
-and CDN three different services within one HTTP/2 connection.
+enables clients to access different services without initiating new connections,
+or including routing metadata in every message.  As shown in the following
+diagram, the client can exchange data with a publish/subscribe service, an RPC
+service and a CDN within one HTTP/2 connection.
```
Comment (erickinnear) [inline draft-xie-bidirectional-messaging-latest.md]:
```suggestion
service, and a CDN within one HTTP/2 connection.
```

**Review by erickinnear — APPROVED** [review]
Overall good stuff! A few minor typo-level nits.

**Comment by erickinnear** [comment]
@afrind no worries! Either order works for me.

**Review by w-o-o — APPROVED** [review]

**Pull request merged**